deepchem/data/datasets.py

```diff
@@ -499,9 +499,9 @@ def generator():
       w_next = np.zeros((0,) + (len(tasks),))
       ids_next = np.zeros((0,), dtype=object)
       for (X, y, w, ids) in self.itershards():
-        X_next = np.vstack([X_next, X])
-        y_next = np.vstack([y_next, y])
-        w_next = np.vstack([w_next, w])
+        X_next = np.concatenate([X_next, X], axis=0)
+        y_next = np.concatenate([y_next, y], axis=0)
+        w_next = np.concatenate([w_next, w], axis=0)
         ids_next = np.concatenate([ids_next, ids])
```

Comment:
When resharding on graph-conv features, X is only an 1d array, using `np.vstack`could lead to dimension error